lib/MatterbridgeManager.php

```diff
@@ -548,6 +560,106 @@ private function checkBridgeProcess(Room $room, array $bridge, bool $relaunch =
 		return $pid;
 	}
 
+	/**
+	 * Send system message to the room if necessary
+	 *
+	 * @param Room $room the room
+	 * @param string $userId the editing user
+	 * @param array $currentBridge previous bridge values
+	 * @param array $newBridge future bridge values
+	 */
+	private function notify(Room $room, string $userId, array $currentBridge, array $newBridge): void {
+		$currentParts = $currentBridge['parts'];
+		$newParts = $newBridge['parts'];
+		if (count($currentParts) === 0 && count($newParts) > 0) {
+			$this->sendSystemBridgeAdded($room, $userId);
+		} elseif (count($currentParts) > 0 && count($newParts) === 0) {
+			$this->sendSystemBridgeRemoved($room, $userId);
+		} elseif (count($currentParts) !== count($newParts) || !$this->compareBridges($currentBridge, $newBridge)) {
+			$this->sendSystemBridgeEdited($room, $userId);
+		}
+	}
+
+	/**
+	 * Check if 2 bridge configurations are identical
+	 *
+	 * @param array $bridge1
+	 * @param array $bridge2
+	 * @return bool True if they are strictly equivalent
+	 */
+	private function compareBridges(array $bridge1, array $bridge2): bool {
+		// try to find an equivalent for each bridge part of one side in the other
+		foreach ($bridge1['parts'] as $part1) {
+			$found = false;
+			foreach ($bridge2['parts'] as $part2) {
+				if ($this->compareBridgeParts($part1, $part2)) {
+					$found = true;
+					break;
+				}
+			}
+			if (!$found) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Check if 2 bridge parts are identical
+	 *
+	 * @param array $part1
+	 * @param array $part2
+	 * @return bool True if they are strictly equivalent
+	 */
+	private function compareBridgeParts(array $part1, array $part2): bool {
+		$keys1 = array_keys($part1);
+		$keys2 = array_keys($part2);
+		if (count($keys1) !== count($keys2)) {
+			return false;
+		}
+
+		foreach ($part1 as $key => $val) {
+			if (!isset($part2[$key]) || $val !== $part2[$key]) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	private function sendSystemBridgeAdded(Room $room, string $userId): void {
```

Comment:
Combine all of those 3 to one and add `'matterbridge_config_added'` as a parameter?